python/equistore-operations/equistore/operations/equal.py

```diff
@@ -56,7 +56,7 @@ def equal_raise(tensor_1: TensorMap, tensor_2: TensorMap):
     except ValueError as e:
         raise NotEqualError("the tensor maps have different keys") from e
 
-    for key, block_1 in tensor_1:
+    for key, block_1 in tensor_1.items():
```

Comment:
Yes, this is a breaking change.

The reason for it is TorchScript does not support `yield`, so we can not define a custom `__iter__`. Instead we get the default Python behavior: if we define `__getitem__`, iterating over the object will call it with increasing integers until it gets an `IndexError`. Since `__getitem__` returns blocks, iterating over a tensor will give us blocks.

Since we still want a way to get both keys and blocks, I've added this `items` functions for it.

Reply:
I see, all of the above makes sense. Thanks!